accepted/future-releases/extension-types/feature-specification.md

```diff
@@ -1110,9 +1110,8 @@ Assume that _DV_ is an extension type declaration named `Name`, and
 `V1` occurs as one of the `<type>`s in the `<interfaces>` of _DV_. In
 this case we say that `V1` is a _superinterface_ of _DV_.
 
-If _DV_ does not include an `<interfaces>` clause then _DV_ has
-`Object?` or `Object` as a direct superinterface, according to the subtype
-relation which was specified earlier.
+_DV_ has `Object?` or `Object` as a direct superinterface, according to the
+subtype relation which was specified earlier.
```

Comment:
My second worry is more nit-picky.

Does it even make sense to say that _DV_, a *declaration*, has a superinterface *type*, direct or otherwise?

In the class modifiers spec, I introduced "declared interface" to distinguish the declaration from the type.
The problem here is also that `Object?` is not an interface. I'd be *much* less worried if we dropped the notion of `Object?` being treated like a nominative type, which can be a superinterface. It's unnecessary, and it makes things more complicated, not less, because it breaks down distinctions that have so far been clear.
(You can't have a structural type as superinterface. Except, now you can? What does that even mean.)

Even without that, with declarations like:
```dart
extension type E1<X extends int?>(X _) {}
extension type E2<X extends int>(E<X> _) {}
```
will `E1` or `E2` have `Object` as superinterface?
(My point being that it's a meaningless question without knowing `X`.)

We can probably say that `E1` does not have `Object` as direct superinterface,
and `E2` does, because `E1<X>` (a type, not a declaration) does have `Object` as supertype.

But what were we even *answering* when we said that `E1` does not have `Object` as direct superinterface?
When is that information needed?

And because of that, should we stop saying that `Object` and `Object?` are superinterfaces of the _declaration_, and make sure to say it only about the *types*, where it's just a superinterface, not a direct superinterface (I think).

Reply:
I think it's fair to say that this has been resolved.